src/main/java/mate/academy/internetshop/dao/impl/UserDaoImpl.java

```diff
@@ -42,4 +42,9 @@ public boolean delete(Long id) {
         return Storage.users
                 .removeIf(user -> user.getId().equals(id));
     }
+
+    @Override
+    public Optional<User> findByLogin(String login) {
+        return Storage.users.stream().filter(s -> s.getLogin().equals(login)).findFirst();
```

Comment:
```suggestion
        return Storage.users.stream()
        .filter(s -> s.getLogin().equals(login))
        .findFirst();
```
не забуваємо за оформлення коду, коли ми юзаєм стріми

Reply:
done